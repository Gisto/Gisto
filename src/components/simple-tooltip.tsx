import { Info } from 'lucide-react';
import { ReactNode } from 'react';

import { Tooltip, TooltipContent, TooltipTrigger } from '@/components/ui/tooltip.tsx';

export const SimpleTooltip = ({
  children,
  content,
  className,
}: {
  children?: ReactNode;
  content: ReactNode;
  className?: string;
}) => {
  const DefaultTrigger = <Info strokeWidth={1.5} className="size-3" />;
  return (
    <Tooltip>
      <TooltipTrigger>{children ?? DefaultTrigger}</TooltipTrigger>
      <TooltipContent className={className}>⚠︎ {content}</TooltipContent>
    </Tooltip>
  );
};
