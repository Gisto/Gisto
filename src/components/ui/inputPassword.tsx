import { Eye, EyeClosed } from 'lucide-react';
import * as React from 'react';
import { useState } from 'react';

import { cn } from '@/lib/utils';

const InputPassword = React.forwardRef<HTMLInputElement, React.ComponentProps<'input'>>(
  ({ className, ...props }, ref) => {
    const [show, setMask] = useState<boolean>(false);

    return (
      <div className={'relative flex items-center gap-2'}>
        <input
          type={show ? 'text' : 'password'}
          className={cn(
            'flex h-10 w-full rounded-md border border-input bg-background px-3 py-2 text-base ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 md:text-sm',
            !show && ' tracking-wide',
            className
          )}
          ref={ref}
          {...props}
        />
        {show ? (
          <Eye className="w-4 h-4 cursor-pointer" onClick={() => setMask(false)} />
        ) : (
          <EyeClosed className="w-4 h-4 cursor-pointer" onClick={() => setMask(true)} />
        )}
      </div>
    );
  }
);
InputPassword.displayName = 'InputPassword';

export { InputPassword };
