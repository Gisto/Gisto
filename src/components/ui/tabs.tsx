'use client';

import * as TabsPrimitive from '@radix-ui/react-tabs';
import * as React from 'react';
import { cloneElement, isValidElement } from 'react';

import { useIsMobile } from '@/hooks/use-mobile';
import { cn } from '@/lib/utils';

interface ChildProps {
  variant?: 'line' | 'pill';
}

const Tabs = TabsPrimitive.Root;
const TabsList = React.forwardRef<
  React.ElementRef<typeof TabsPrimitive.List>,
  React.ComponentPropsWithoutRef<typeof TabsPrimitive.List> & ChildProps
>(({ className, ...props }, ref) => {
  const isMobile = useIsMobile();

  const childrenWithEnhancedProps = React.Children.map(props.children, (child) => {
    if (isValidElement(child)) {
      const childProps = child.props as ChildProps;
      return cloneElement(child as React.ReactElement<ChildProps>, {
        ...childProps,
        variant: props.variant,
      });
    }

    return child;
  });

  return (
    <TabsPrimitive.List
      ref={ref}
      className={cn(
        'inline-flex h-10 items-center justify-center rounded-md bg-muted p-1 text-muted-foreground',
        props?.variant === 'line' &&
          'inline-flex w-full !items-center !justify-start !p-0 bg-transparent rounded-none gap-0 border-b-2 border-border text-muted-foreground mb-4',
        props?.variant !== 'line' &&
          isMobile &&
          'flex items-center justify-start flex-wrap h-auto space-y-1',
        className
      )}
      {...props}
    >
      {childrenWithEnhancedProps}
    </TabsPrimitive.List>
  );
});
TabsList.displayName = TabsPrimitive.List.displayName;

const TabsTrigger = React.forwardRef<
  React.ElementRef<typeof TabsPrimitive.Trigger>,
  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Trigger> & {
    variant?: 'line' | 'pill';
  }
>(({ className, ...props }, ref) => {
  return (
    <TabsPrimitive.Trigger
      ref={ref}
      className={cn(
        'inline-flex items-center justify-center whitespace-nowrap rounded-sm px-3 py-1.5 text-sm font-medium ring-offset-background transition-all focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 data-[state=active]:bg-background data-[state=active]:text-foreground data-[state=active]:shadow-sm',
        props?.variant === 'line' &&
          'mx-2 -mb-[2px] rounded-none shadow-none inline-flex items-center justify-start  whitespace-nowrap border-b-2 px-2 py-2 text-base font-medium transition-all first-of-type:ml-0 disabled:pointer-events-none disabled:text-muted-foreground data-[state=active]:border-primary data-[state=inactive]:border-transparent data-[state=active]:shadow-none data-[state=active]:font-semibold data-[state=active]:text-foreground',
        className
      )}
      {...props}
    />
  );
});
TabsTrigger.displayName = TabsPrimitive.Trigger.displayName;

const TabsContent = React.forwardRef<
  React.ElementRef<typeof TabsPrimitive.Content>,
  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Content>
>(({ className, ...props }, ref) => (
  <TabsPrimitive.Content
    ref={ref}
    className={cn(
      'mt-2 ring-offset-background focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2',
      className
    )}
    {...props}
  />
));
TabsContent.displayName = TabsPrimitive.Content.displayName;

export { Tabs, TabsList, TabsTrigger, TabsContent };
