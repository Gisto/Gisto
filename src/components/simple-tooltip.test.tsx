import { render } from '@testing-library/react';

import { SimpleTooltip } from './simple-tooltip';

import { TooltipProvider } from '@/components/ui/tooltip';

describe('SimpleTooltip Component', () => {
  it('renders correctly with default trigger', () => {
    const { asFragment } = render(
      <TooltipProvider>
        <SimpleTooltip content="Tooltip content" />
      </TooltipProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with custom trigger', () => {
    const { asFragment } = render(
      <TooltipProvider>
        <SimpleTooltip content="Tooltip content">
          <a>Custom Trigger</a>
        </SimpleTooltip>
      </TooltipProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('applies custom className to TooltipContent', () => {
    const { asFragment } = render(
      <TooltipProvider>
        <SimpleTooltip content="Tooltip content" className="custom-class" />
      </TooltipProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
