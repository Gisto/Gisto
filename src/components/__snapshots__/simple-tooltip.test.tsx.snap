// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SimpleTooltip Component > applies custom className to TooltipContent 1`] = `
<DocumentFragment>
  <button
    data-state="closed"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-info size-3"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
  </button>
</DocumentFragment>
`;

exports[`SimpleTooltip Component > renders correctly with custom trigger 1`] = `
<DocumentFragment>
  <button
    data-state="closed"
  >
    <a>
      Custom Trigger
    </a>
  </button>
</DocumentFragment>
`;

exports[`SimpleTooltip Component > renders correctly with default trigger 1`] = `
<DocumentFragment>
  <button
    data-state="closed"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-info size-3"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="1.5"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        r="10"
      />
      <path
        d="M12 16v-4"
      />
      <path
        d="M12 8h.01"
      />
    </svg>
  </button>
</DocumentFragment>
`;
